src/main.rs

```diff
@@ -211,6 +218,13 @@ async fn withdraw_accumulated_cycles(provider_id: u64, canister_id: Principal) {
         provider.cycles_owed = 0;
         p.borrow_mut().insert(provider_id, provider)
     });
+    log!(
```

Comment:
Would it make sense to also log the caller? This could help to track that it was indeed the owner of the provider.

Reply:
:heavy_plus_sign:

Reply:
Good point; updated.